### PR TITLE
[TECHNICAL] Kotlinize PassCodeManager

### PR DESCRIPTION
**Pull request by JuancaG05**
## Related Issues
App: https://github.com/owncloud/android/issues/3371
_____

### CODE REVIEW

**Review by fesave — APPROVED** [review]
LGTM

File: owncloudApp/src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.kt [hunk 1]
```diff
@@ -519,7 +519,7 @@ class FileDisplayActivity : FileActivity(), FileFragment.ContainerActivity, OnEn
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
             BiometricManager.getBiometricManager(this).bayPassUnlockOnce()
         }
-        PassCodeManager.getPassCodeManager().bayPassUnlockOnce()
+        PassCodeManager.getPassCodeManager()!!.bayPassUnlockOnce()
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.kt:522]: Try not to add !! here 👍

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/security/PassCodeManager.kt [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import android.os.PowerManager
+import android.os.SystemClock
+import android.preference.PreferenceManager
+import com.owncloud.android.MainApp.Companion.appContext
+import com.owncloud.android.authentication.BiometricManager
+
+object PassCodeManager {
+
+    var exemptOfPasscodeActivites: MutableSet<Class<*>> = mutableSetOf(PassCodeActivity::class.java)
+        private set
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/security/PassCodeManager.kt:36]: Does it really need to be a `var` with a `private set`? I mean, is it used out of the passcode manager? Could it be a `private val`?
```suggestion
    private val exemptOfPasscodeActivities: MutableSet<Class<*>> = mutableSetOf(PassCodeActivity::class.java)
```

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/security/PassCodeManager.kt [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+    private var timestamp = 0L
+    private var visibleActivitiesCounter = 0
+
+    // keeping a "low" positive value is the easiest way to prevent the pass code is requested on rotations
+    private val PASS_CODE_TIMEOUT = 1000
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/security/PassCodeManager.kt:42]: This one could be marked as a constant, couldn't it?
```suggestion
    private const val PASS_CODE_TIMEOUT = 1_000
```

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/security/PassCodeManager.kt [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    fun isPassCodeEnabled(): Boolean {
+        val appPrefs = PreferenceManager.getDefaultSharedPreferences(appContext)
+        return appPrefs.getBoolean(PassCodeActivity.PREFERENCE_SET_PASSCODE, false)
+    }
+
+    private fun checkPasscode(activity: Activity) {
+        val i = Intent(appContext, PassCodeActivity::class.java)
+        i.action = PassCodeActivity.ACTION_CHECK
+        i.flags = Intent.FLAG_ACTIVITY_REORDER_TO_FRONT or Intent.FLAG_ACTIVITY_SINGLE_TOP
+        activity.startActivity(i)
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/security/PassCodeManager.kt:88]: Let's kotlinize it a little bit more 😉 
```suggestion
        val intent = Intent(appContext, PassCodeActivity::class.java).apply {
            action = PassCodeActivity.ACTION_CHECK
            flags = Intent.FLAG_ACTIVITY_REORDER_TO_FRONT or Intent.FLAG_ACTIVITY_SINGLE_TOP
        }
        activity.startActivity(intent)
    }
```

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/security/PassCodeManager.kt [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        if (isPassCodeEnabled() && !powerMgr.isScreenOn) {
+            activity.moveTaskToBack(true)
+        }
+    }
+
+    // other activities may be exempted, if needed
+    fun addExemptOfPasscodeActivity(activity: Class<*>) {
+        exemptOfPasscodeActivites.add(activity)
+    }
+
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/security/PassCodeManager.kt:73]: If it is not needed at the moment, I think we can remove this. We can add it when we need it again, what do you think?

**Review by abelgardep — CHANGES_REQUESTED** [review]
Last suggestions before moving it forward @JuancaG05 👍

**Review by abelgardep — APPROVED** [review]
LGTM 💯

**Comment by abelgardep** [comment]
Will be tested within https://github.com/owncloud/android/pull/3361

**Pull request merged**